### PR TITLE
update java mgmt code sample

### PR DESCRIPTION
**Pull request by xccc-msft**
As `azure-resourcemanager:2.0.0` reaches GA, this PR is to update the code sample related to management Java SDK.

1. Remove auth file related part. 
Add EnvionmentCredential as recommended setting for the sample project. By the end, link to azure identity to introduce more authentication implementations if required.

2. Add new dependencies for running the sample.
Add `azure-identity` and `azure-resourcemanager` for GA version.
Please note that additional `azure-resourcemanager-sql` is added with comment as it's still in preview.

3. Migrate code sample.
Update the samples with above dependencies. After SQL reaches GA, we might need to remove the preview dependency and sync this doc again.

### CODE REVIEW

**Review by nickzhums — APPROVED** [review]

File: articles/java/sdk/java-sdk-azure-get-started.md [hunk 1]
```diff
@@ -45,40 +45,14 @@ Which gives you a reply in the following format:
...
-```bash
-export AZURE_AUTH_LOCATION=/Users/raisa/azureauth.properties
-```
-
-If you're working in a windows environment, add the variable to your system properties. Open a PowerShell window with administrator privleges and, after replacing the second variable with the path to your file, enter the following command:
-
-```powershell
-setx AZURE_AUTH_LOCATION "C:\<fullpath>\azureauth.properties" /m
-```
+For more approaches of authentication support, please refer to [Azure Identity](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/identity/azure-identity#azure-identity-client-library-for-java).
```
Comment (weidongxu-microsoft) [inline articles/java/sdk/java-sdk-azure-get-started.md:55]: "For more approaches of authentication support" seems not fit here. suggest "For more options of authentication" or "For support on authentication".

File: articles/java/sdk/java-sdk-azure-get-started.md [hunk 2]
```diff
@@ -248,15 +225,22 @@ Replace the main method in `AzureApp.java` with the one below, updating the `app
...
+            TokenCredential credential = new EnvironmentCredentialBuilder()
+                    .authorityHost(AzureAuthorityHosts.AZURE_PUBLIC_CLOUD)
+                    .build();
+
+            // if you do not set tenant ID and subscription ID via environment variables
+            // change to create azure profile with tenantId, subscriptionId and azure environment
+            AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+            
+            AzureResourceManager azureResourceManager = AzureResourceManager.configure()
+                    .withLogLevel(HttpLogDetailLevel.BASIC.BASIC)
```
Comment (weidongxu-microsoft) [inline articles/java/sdk/java-sdk-azure-get-started.md:239]: what's this? make sure the code at least compiles.

Comment (xccc-msft) [reply articles/java/sdk/java-sdk-azure-get-started.md]: It's copy-paste issue here. As no compile error would throw, I just missed it. Fixed now.

File: articles/java/sdk/java-sdk-azure-get-started.md (hunk 1 above)
Comment (xccc-msft) [reply articles/java/sdk/java-sdk-azure-get-started.md]: Updated by `For more options of authentication`.

**Comment by ktoliver** [comment]
@rloutlaw Could you review the proposed update to your article? Thanks.

**Comment by KarlErickson** [comment]
@xccc-msft could you please resubmit this PR in the private repo (azure-dev-docs-pr)? Internal people should always use the internal repo. Thanks!

**Comment by xccc-msft** [comment]
Close this as it is expected to be in private repo.

**Pull request closed without merging**